lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2013,17 +2002,20 @@ static void checkAccessibility(TypeChecker &TC, const Decl *D) {
       }
     });
 
-    if (minAccessScope) {
-      auto minAccess = accessibilityFromScopeForDiagnostics(minAccessScope);
+    if (!minAccessScope.isPublic()) {
+      auto minAccess = minAccessScope.accessibilityForDiagnostics();
       bool isExplicit =
         SD->getAttrs().hasAttribute<AccessibilityAttr>() ||
         SD->getDeclContext()->getAsProtocolOrProtocolExtensionContext();
       auto diagID = diag::subscript_type_access;
       if (downgradeToWarning == DowngradeToWarning::Yes)
         diagID = diag::subscript_type_access_warn;
+      auto subscriptDeclAccess = isExplicit
+        ? SD->getFormalAccess()
+        : minAccessScope.requiredAccessibilityForDiagnostics();
```

Comment:
:-/ I guess I'm unhappy because the same diagnostic argument is being used for two different purposes now, which makes me think there should just be two separate diagnostics. We can fix that in a follow-up commit, though.